packages/api-client/src/granules.ts

```diff
@@ -442,16 +495,25 @@ export const createGranule = async (params: {
 export const updateGranule = async (params: {
   prefix: string,
   body: ApiGranule,
+  granuleId: GranuleId,
+  collectionId?: CollectionId,
   callback?: InvokeApiFunction
 }): Promise<ApiGatewayLambdaHttpProxyResponse> => {
-  const { prefix, body, callback = invokeApi } = params;
+  const { prefix, granuleId, collectionId, body, callback = invokeApi } = params;
+
+  let path = `/granules/${collectionId}/${granuleId}`;
+
+  // Fetching a granule without a collectionId is supported but deprecated
+  if (!collectionId) {
+    path = `/granules/${granuleId}`;
+  }
 
   return await callback({
     prefix,
     payload: {
       httpMethod: 'PUT',
       resource: '/{proxy+}',
-      path: `/granules/${body.granuleId}`,
+      path,
```

Comment:
This change is noteworthy. For some reason the `updateGranule` api-client method here was setting the `path` using the `body.granuleId`, rather than a passed-in param. This means that we weren't testing the logic matching the query parameter to the body's granuleId. It also meant that all of the calls to `updateGranule` needed to be updated to pass in the `granuleId` param, hence the integration test updates here.

Reply:
👍